library/src-bootstrapped/scala/tasty/reflect/TreeUtils.scala

```diff
@@ -157,7 +157,7 @@ trait TreeUtils
       }
     }
 
-    def transformTerm(tree: Term)(given ctx: Context): Term = {
+    def transformTerm(tree: Term)(given ctx: Context): Term = if IsTerm.unapply(tree).isEmpty then tree else
```

Comment:
This isn't correct, if `tree` is a `Term` then `IsTerm.unapply(tree)` should never be empty. Where did that tree come from?

Reply:
The reason for this is (at least) the `UnApply` type. Technically it is `<: Term` but for the purposes of Tasty reflect we do not treat it as such since it cannot appear in a term position.

Reply:
We discussed earlier the failing example for this:

Macro:
```scala
import scala.quoted.{ given, _ }

inline def mcr(x: => Unit): Unit = ${mcrImpl('x)}
def mcrImpl(x: Expr[Unit])(given ctx: QuoteContext): Expr[Unit] =
  import ctx.tasty.{ given, _ }
  val tr: Term = x.unseal
  object m extends TreeMap
  m.transformTerm(tr).seal.cast[Unit]
```

Test:
```scala
@main def Test = mcr {
  val (a, b) = ???
}
```